src/popover/container.tsx

```diff
@@ -191,8 +191,8 @@ export default function PopoverContainer({
 
     const onClick = (event: UIEvent | KeyboardEvent) => {
       if (
-        // Do not update position if the click was on the popover itself and keepPosition is true.
-        (keepPosition && nodeContains(popoverRef.current, event.target)) ||
+        // Do not update position if keepPosition is true.
+        keepPosition ||
```

Comment:
Simplified this condition to not check whether the event was in the popover, because when dismissing, there is a frame before where the dismiss button disappears while the popover is still visible. This would cause `nodeContains(popoverRef.current, event.target)` to be false and update the popover position undesirably for that one frame before disappearing.

For the use case that uses the `keepPosition` prop, we don't need to consider clicks outside the popover for resizing because such clicks would close it anyway.

Reply:
Do we still need to check for target? I think that is merely a performance optimisation which looks unnecessary

Reply:
True that it's not that related to the PR, but it prevents unnecessary recalculations and re-renders.